src/pwm_input.rs

```diff
@@ -4,48 +4,37 @@
 use core::marker::PhantomData;
 use core::mem;
 
-use crate::pac::{DBGMCU as DBG, TIM2, TIM3};
+use crate::pac::DBGMCU as DBG;
+#[cfg(any(
+    feature = "stm32f100",
+    feature = "stm32f103",
+    feature = "stm32f105",
+))]
+use crate::pac::TIM1;
+use crate::pac::{TIM2, TIM3};
 #[cfg(feature = "medium")]
 use crate::pac::TIM4;
 
 use crate::afio::MAPR;
-use crate::gpio::gpioa::{PA0, PA1, PA15, PA6, PA7};
-use crate::gpio::gpiob::{PB3, PB4, PB5};
-#[cfg(feature = "medium")]
-use crate::gpio::gpiob::{PB6, PB7};
-use crate::gpio::{Floating, Input};
+use crate::gpio::{self, Floating, Input};
 use crate::rcc::Clocks;
 use crate::time::Hertz;
 use crate::timer::Timer;
 
-pub trait Pins<TIM> {
-    const REMAP: u8;
-}
-
-#[cfg(feature = "medium")]
-impl Pins<TIM4> for (PB6<Input<Floating>>, PB7<Input<Floating>>) {
-    const REMAP: u8 = 0b0;
-}
-
-impl Pins<TIM3> for (PA6<Input<Floating>>, PA7<Input<Floating>>) {
-    const REMAP: u8 = 0b00;
-}
+pub trait Pins<REMAP> {}
 
-impl Pins<TIM3> for (PB4<Input<Floating>>, PB5<Input<Floating>>) {
-    const REMAP: u8 = 0b10;
-}
+use crate::timer::sealed::{Remap, Ch1, Ch2};
 
-impl Pins<TIM2> for (PA0<Input<Floating>>, PA1<Input<Floating>>) {
-    const REMAP: u8 = 0b00;
-}
-
-impl Pins<TIM2> for (PA15<Input<Floating>>, PB3<Input<Floating>>) {
-    const REMAP: u8 = 0b11;
-}
+impl<TIM, REMAP, P1, P2> Pins<REMAP> for (P1, P2)
+where
+    REMAP: Remap<Periph = TIM>,
+    P1: Ch1<REMAP> + gpio::Mode<Input<Floating>>,
+    P2: Ch2<REMAP> + gpio::Mode<Input<Floating>> {}
```

Comment:
@geomatsi pwm_input use other `Pins` trait/implementation with only first two channels. I did not change this.